### PR TITLE
docs: update readme for brevity and to add more info about WebAssembly, macOS, and Windows

### PR DESCRIPTION
**Pull request by deadprogram**
This PR is to update the README to make it a bit shorter, since there are so many MCU boards. It also is to bring WebAssembly support a bit more into the spotlight.

### CODE REVIEW

**Comment by deadprogram** [comment]
Note that it references some sections in the tinygo site that do not exist yet, or need updates.

**Comment by deadprogram** [comment]
The tinygo-site `dev` branch contains the updates now, so this can be reviewed/merged/etc.

File: README.md [hunk 1]
```diff
@@ -35,115 +37,61 @@ The above program can be compiled and run without modification on an Arduino Uno
...
+
+TinyGo programs can run in Fastly Compute@Edge (https://developer.fastly.com/learning/compute/go/), Fermyon Spin (https://developer.fermyon.com/spin/go-components), wazero (https://wazero.io/languages/tinygo/) and many other WebAssembly runtimes.
+
+Here is a small TinyGo program for use by a WASI host application:
+
+```go
+package main
+
+//export add
+func add(x, y uint32) uint32 {
```
Comment (aykevl) [inline README.md:53]: Same as my comment here: https://github.com/tinygo-org/tinygo-site/pull/349#discussion_r1178105409

File: README.md [hunk 2]
```diff
@@ -35,115 +37,61 @@ The above program can be compiled and run without modification on an Arduino Uno
...
-* [Waveshare RP2040-Zero](https://www.waveshare.com/wiki/RP2040-Zero)
-* [X9 Pro smartwatch](https://github.com/curtpw/nRF5x-device-reverse-engineering/tree/master/X9-nrf52832-activity-tracker/)
-
-
-For more information, see [this list of boards](https://tinygo.org/microcontrollers/). Pull requests for additional support are welcome!
+## Supported targets
+
+### Embedded
+
+You can compile TinyGo programs for over 94 different microcontrollers.
```
Comment (aykevl) [inline README.md]: Microcontroller boards, not microcontrollers. The atsamd21g18a is supported on ~9 boards, but it's still the same microcontroller in each case.

**Review by aykevl — COMMENTED** [review]
Overall this seems like a good improvement. Let's not keep information about supported boards in two different places.

File: README.md (hunk 1 above)
Comment (deadprogram) [reply README.md:53]: OK updated.

File: README.md (hunk 2 above)
Comment (deadprogram) [reply README.md]: Corrected.

**Review by aykevl — APPROVED** [review]
LGTM

**Comment by deadprogram** [comment]
Thanks for review @aykevl now merging.

**Pull request merged**